File: src/ios/Fragment.swift
//
//  Fragment.swift
//  StepCounterTest
//
//  Created by Leo on 3/6/19.
//  Copyright © 2019 Leonard Greulich. All rights reserved.
//

struct Fragment {
    enum orders {
        case MaxMinMax
        case MinMaxMin
        case none
    }
    
    let heightMax: Double
    let heightMin: Double
    let amplitude: Double
    let lengthFirst: Int
    let lengthSecond: Int
    let lengthTotal: Int
    let axis: Int
    let fragmentType: orders
    
    init() {
        self.heightMax = 0
        self.heightMin = 0
        self.amplitude = 0
        self.lengthFirst = 0
        self.lengthSecond = 0
        self.lengthTotal = 0
        self.axis = 0
        self.fragmentType = .none
    }
    
    init(_ heightMax: Double, _ heightMin: Double, _ lengthFirst: Int, _ lengthSecond: Int, _ axis: Int, _ fragmentType: orders) {
        self.heightMax = heightMax
        self.heightMin = heightMin
        self.amplitude = heightMax - heightMin
        self.lengthFirst = lengthFirst
        self.lengthSecond = lengthSecond
        self.lengthTotal = lengthFirst + lengthSecond
        self.axis = axis
        self.fragmentType = fragmentType
    }
    
    init(_ amplitude: Double, _ lengthTotal: Int, _ axis: Int, _ fragmentType: orders) {
        self.heightMax = 0
        self.heightMin = 0
        self.amplitude = amplitude
        self.lengthFirst = 0
        self.lengthSecond = 0
        self.lengthTotal = lengthTotal
        self.axis = axis
        self.fragmentType = fragmentType
    }
}

File: src/android/Fragment.java
package cordova.plugin.stepdist;

public class Fragment {

enum orders {
    MaxMinMax, MinMaxMin, none
}

    Double heightMax;
    Double heightMin;
    Double amplitude;
    Integer lengthFirst;
    Integer lengthSecond;
    Integer lengthTotal;
    Integer axis;
    orders fragmentType;

    public Fragment() {
        this.heightMax = 0.0;
        this.heightMin = 0.0;
        this.amplitude = 0.0;
        this.lengthFirst = 0;
        this.lengthSecond = 0;
        this.lengthTotal = 0;
        this.axis = 0;
        this.fragmentType = orders.none;
    }

    public Fragment(Double heightMax, Double heightMin, Integer lengthFirst, Integer lengthSecond, Integer axis, orders fragmentType) {
        this.heightMax = heightMax;
        this.heightMin = heightMin;
        this.amplitude = heightMax - heightMin;
        this.lengthFirst = lengthFirst;
        this.lengthSecond = lengthSecond;
        this.lengthTotal = lengthFirst + lengthSecond;
        this.axis = axis;
        this.fragmentType = fragmentType;
    }

    public Fragment(Double amplitude, Integer lengthTotal, Integer axis, orders fragmentType) {
        this.heightMax = 0.0;
        this.heightMin = 0.0;
        this.amplitude = amplitude;
        this.lengthFirst = 0;
        this.lengthSecond = 0;
        this.lengthTotal = lengthTotal;
        this.axis = axis;
        this.fragmentType = fragmentType;
    }
}
